### PR TITLE
Update ContentDataSource.java

### PR DESCRIPTION
**Pull request by axet**
Checking File size makes more sense then asking for InputStream cached bytes.

#3426 explanation

### CODE REVIEW

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ public long open(DataSpec dataSpec) throws ContentDataSourceException {
         long assetFileDescriptorLength = assetFileDescriptor.getLength();
         if (assetFileDescriptorLength == AssetFileDescriptor.UNKNOWN_LENGTH) {
           // The asset must extend to the end of the file.
-          bytesRemaining = inputStream.available();
+          bytesRemaining = ((FileInputStream) inputStream).getChannel().size();
           if (bytesRemaining == 0) {
             // FileInputStream.available() returns 0 if the remaining length cannot be determined,
```
Comment (ojw28) [inline library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:94]: This comment doesn't seem to make sense given then change being made? Does the if block itself still make sense?

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java [hunk 2]
```diff
@@ -89,7 +89,7 @@ public long open(DataSpec dataSpec) throws ContentDataSourceException {
         long assetFileDescriptorLength = assetFileDescriptor.getLength();
         if (assetFileDescriptorLength == AssetFileDescriptor.UNKNOWN_LENGTH) {
           // The asset must extend to the end of the file.
-          bytesRemaining = inputStream.available();
+          bytesRemaining = ((FileInputStream) inputStream).getChannel().size();
```
Comment (ojw28) [inline library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:92]: I think it's possible for multiple assets to be in a single file. In which case is `size()` the size of just this one, or of all of them? I think you might want to subtract `position()` of the `FileChannel` to get correct behavior?

**Comment by ojw28** [comment]
* We wont be able to merge this without a signed CLA.
* Pull requests need to go to the `dev-v2` branch, not `release-v2`. Please address the comments and make a new request to that branch.

**Comment by axet** [comment]
1) Will wont be able to merge this without a signed CLA.

less then 5 lines of code need to sign anything? Eclipse treat small bug fixes wihtout any paper work. I do not wana bother to sing anything.

2) Ok, I close this one, but keep reffered issue open.

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java (hunk 2 above)
Comment (axet) [reply library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:92]: Not sure what do you mean, FileChannel one per file.

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java (hunk 1 above)
Comment (axet) [reply library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:94]: bytesRemaining = inputStream.available()

- for the momen when you open Pipe this call can return 0. 

bytesRemaining = ((FileInputStream) inputStream).getChannel().size();

- FileDescription for correct file in file system, this call will return correct file size. 

Again. I do not understand, why you asking this. Propbably need more testing.

**Comment by ojw28** [comment]
Please do not send a second pull request if you don't intend to sign the CLA. We wont be able to accept it. Thanks.

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java (hunk 2 above)
Comment (ojw28) [reply library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:92]: I don't think the `uri` for which the source is being opened necessarily corresponds to the whole file though. Else this code wouldn't need to be using `assetFileDescriptor.getStartOffset` as above? If the `uri` only corresponds to a part of the file, that needs to be accounted for.

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java (hunk 1 above)
Comment (ojw28) [reply library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:94]: In the previous code we actually called `inputStream.available()` and then did something conditionally on the value being 0. We're not making an `inputStream.available()` call any more, so the comment doesn't make sense. Can `size()` return 0? If not why is this if block still here? If so the comment needs changing.

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java (hunk 2 above)
Comment (ojw28) [reply library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:92]: Even if the `uri` does correspond to the whole file, you'd need to subtract `skipped` if it's non-zero.

Comment (axet) [reply library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java:92]: Then correct line is:

bytesRemaining = ((FileInputStream) inputStream).getChannel().size() - assetFileDescriptor.getStartOffset

**Comment by axet** [comment]
@ojw28 I didnt know you trying to use "documents" to protect you from the world. I would'nt pull again.

Anyway you can't protect yourself using this technique sign - and enter. I can prove I made this code having it before the code come to the repository, and since it get there without my sign it be equals staling code. So, sign or no sign - always possible to make you/google guilty and violating copyright slavery... Just make it simple, opensource and fun. BTW i'm not going to sue anyway. Just show you how stupid "CLA" is.

**Pull request closed without merging**